sdks/java/io/amazon-web-services2/build.gradle

```diff
@@ -39,6 +39,7 @@ dependencies {
   compile library.java.aws_java_sdk2_sdk_core
   compile library.java.aws_java_sdk2_sns
   compile library.java.aws_java_sdk2_sqs
+  compile library.java.aws_java_sdk2_sts
```

Comment:
Why this dependency is needed here?

Reply:
The STS dependency is needed for authenticating from EKS with IAM for service accounts. Without it, requests to S3 will give access denied errors.

Reply:
Do you mean only EKS authenticated requests or others too?

Reply:
Any deployment that needs to authenticate by assuming a role using the `AWS_ROLE_ARN` and `AWS_WEB_IDENTITY_TOKEN_FILE` environment variables would need the dependency, although to my knowledge EKS with IAM roles for service accounts is the main use case.

Reply:
This should be a runtime dependency, since it is not needed at compile time. (This is causing Java precommit to fail.)